brut.apktool/apktool-lib/src/main/java/brut/androlib/res/decoder/AXmlResourceParser.java

```diff
@@ -852,7 +852,7 @@ private final void resetEventInfo() {
     private final void doNext() throws IOException {
         // Delayed initialization.
         if (m_strings == null) {
-            m_reader.skipCheckInt(CHUNK_AXML_FILE);
+            m_reader.skipCheckInt(CHUNK_AXML_FILE, CHUNK_AXML_FILE_BROKEN);
```

Comment:
Using `skipCheckChunkTypeInt(CHUNK_AXML_FILE, CHUNK_AXML_FILE_BROKEN)` doesn't work.

During the first call to `skipCheckChunkTypeInt()` (though after reading `got`) you would end up with `got=0x00080001`, `expected=0x00080003` and `possible=0x00080001`. because `got == possible` we end up with with a second call and thus a second read of a 32 bit field, where you have `got=0x000109a4`, `expected=0x00080003` and `possible=0xffffffff`.

Reversing the two arguments to `skipCheckChunkTypeInt(CHUNK_AXML_FILE_BROKEN, CHUNK_AXML_FILE)` causes a number of tests to fail. I'm assuing that the test files start with `0x00080003 as expected. This means that during the first call to `skipCheckChunkTypeInt()` you end up with `got=0x00080003`, `expected=0x00080001` and `possible=0x00080003` causing a second call to `skipCheckChunkTypeInt()` and a second read of where you have `got=0x????????`, `expected=0x00080001`, `possible=0xffffffff.

This means that in both cases we end up reading more data than we want when reading the chunk type. Also `skipCheckChunkTypeInt()` is called by `StringBlock` so I'd like to avoid changing it.